### PR TITLE
Try adding EF

### PR DESCRIPTION
**Pull request by Pilchie**
Attempt to add a simple EF db to the project with SqlLite.

Fails right now because `dotnet ef migrations list` fails to find Newtonsoft.Json.NET.

### CODE REVIEW

File: project.json [hunk 1]
```diff
@@ -22,19 +23,33 @@
...
         }
     },
 
     "tools": {
+        "dotnet-aspnet-codegenerator": "1.0.0-*",
+        "dotnet-ef": {
+            "version": "1.0.0-*",
+            "imports": [
+                "dnxcore50",
+                 "portable-net45+win8"
```
Comment (natemcmaster) [inline project.json]: Does it work if you do `"portable-net452+win81"`?

**Comment by natemcmaster** [comment]
@Pilchie as you know, dotnet-ef is still not 100% working. See https://github.com/aspnet/EntityFramework/issues/3925 and https://github.com/aspnet/EntityFramework/issues/4577. I have a working PR open for partial fixes https://github.com/aspnet/EntityFramework/pull/5002

File: project.json (hunk 1 above)
Comment (Pilchie) [inline project.json]: Progress!

Now I get: 

> The project 'SquashNext' doesn't target a framework compatible with .NET Standard App 1.5. You must target a compatible framework such as 'netstandard1.3' in order to use the Entity Framework .NET Core CLI Commands.

It seems EF isn't aware of netcoreapp yet?

Comment (natemcmaster) [inline project.json]: > It seems EF isn't aware of netcoreapp yet?

Correct. Unless you're using a custom build, current versions of EF don't work on netcoreapp1.0 yet. Try changing your tfm to netstandard1.5 for now as a workaround.

**Comment by Pilchie** [comment]
@natemcmaster I pushed another commit, but now it fails to load Newtonsoft.Json again :-(

```
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
   at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider.Load(Stream stream)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load()
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at HelloMvc.Startup..ctor(IHostingEnvironment env)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Extensions.Internal.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.Design.Internal.StartupInvoker.Invoke(Type type, String[] methodNames, IServiceCollection services)
   at Microsoft.EntityFrameworkCore.Design.Internal.StartupInvoker.ConfigureServices()
   at Microsoft.EntityFrameworkCore.Design.DbContextOperations..ctor(ILoggerProvider loggerProvider, Assembly assembly, Assembly startupAssembly, String environment, String startupProjectDir)
   at Microsoft.EntityFrameworkCore.Design.MigrationsOperations..ctor(AssemblyLoader assemblyLoader, ILoggerProvider loggerProvider, Assembly assembly, Assembly startupAssembly, String environment, String projectDir, String startupProjectDir, String rootNamespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.<>c__DisplayClass4_0.<.ctor>b__5()
   at Microsoft.EntityFrameworkCore.Internal.LazyRef`1.get_Value()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.GetMigrationsImpl(String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass4_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Could not load file or assembly 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

**Comment by natemcmaster** [comment]
@Pilchie which version of dotnet-ef are you using?

**Comment by Pilchie** [comment]
I have `1.0.0-*` in my project.json.  From my project.lock.json, it looks like that is `dotnet-ef/1.0.0-rc3-20555`

**Comment by natemcmaster** [comment]
Does your project run? This newstonsoft.json error is coming from HelloMvc.Startup, not from within dotnet-ef.

**Comment by Pilchie** [comment]
The master branch runs, I haven't tried it in this PR, because I will need to create the migration to get the DB running.  Will experiment some more tonight.

**Comment by Pilchie** [comment]
This branch also runs.  I suspect the issue here is that when I run `dotnet ef migrations list`, something in the EF code loads Newtonsoft.Json version 7.0.0, and then later when the app does something that tries to load version 8.0.0, it fails, because it finds v7 already loaded in the same context.  (Note the error in this case has `The located assembly's manifest definition does not match the assembly reference.`, not `The system cannot find the file specified.`

**Comment by natemcmaster** [comment]
> This branch also runs. I suspect the issue here is that when I run dotnet ef migrations list, something in the EF code loads Newtonsoft.Json version 7.0.0, and then later when the app does something that tries to load version 8.0.0.

This may be due to how dotnet-ef loads user's projects. We're digging into this as a part of https://github.com/aspnet/EntityFramework/issues/4577

**Comment by Pilchie** [comment]
Great!  Let me know if there are any workarounds as you dig in.

**Comment by Pilchie** [comment]
Tag @davidfowl - Is it true that there is a separate graph for each tool, that is independent of the graph for the project itself?  If so, that sounds like it will be problematic for tools like this that want to load project code, as they won't end up unifying on a version, and you'll run into problems like this where the app wants Json.NET v8, but the tool (EF in this case) already has Json.NET v7 loaded.

**Comment by natemcmaster** [comment]
> Is it true that there is a separate graph for each tool, that is independent of the graph for the project itself?

Yes. This is part of the reason why dotnet-razor-tooling, dotnet-ef, and others will start requiring a node in "dependencies", not just tools. We started this move in https://github.com/aspnet/EntityFramework/pull/4977, but have discovered that even this separation isn't enough. Expect more changes before RC2 lands.

**Comment by natemcmaster** [comment]
Btw, I submitted a change for working with a project that uses json.net. This may take a while to run the CI, but the fix should temporarily unblock this. https://github.com/aspnet/EntityFramework/pull/5082

**Comment by natemcmaster** [comment]
@Pilchie we finally pushed the changes to EF to fix a bunch of the bugs you were hitting. We renamed packages too. See https://github.com/aspnet/EntityFramework/issues/3925#issue-119549607 for a sample project.json

**Comment by Pilchie** [comment]
Thanks!  I haven't had time to try much, but at least `dotnet ef migrations list` doesn't crash with the current state of this PR.  Will try to experiment more soon.

**Pull request merged**